### PR TITLE
Fix validation of inputs that aren't text

### PR DESCRIPTION
**Pull request by RobinMcCorkell**
This caused storages containing a checkbox input to not get checked when loading the settings, since `.val()` on a checkbox returns an empty string, and the JS thinks the storage is missing non-optional parameters. It also paves the way for better validation of other input types.

To test:
1. Create a storage with some checkbox parameters (inception ownCloud is suitable)
2. Configure storage and notice the green dot
3. Reload the page, notice no status indicator of any kind
4. With this PR, green dot appears again on page load

cc @PVince81 @icewind1991

### CODE REVIEW

File: apps/files_external/js/settings.js [hunk 1]
```diff
@@ -59,10 +59,24 @@ function highlightBorder($element, highlight) {
 	return highlight;
 }
 
+function isInputValid($input) {
+	optional = $input.hasClass('optional');
```
Comment (PVince81) [inline apps/files_external/js/settings.js]: `var optional`.
Else it's global :scream:

**Comment by PVince81** [comment]
Code looks good otherwise, a unit test would be nice :smile:

File: apps/files_external/js/settings.js (hunk 1 above)
Comment (Xenopathic) [inline apps/files_external/js/settings.js]: Haven't done JS in a while :laughing: Fixed now

**Comment by davitol** [comment]
:+1: WFM

**Comment by Xenopathic** [comment]
@PVince81 Tests!

**Comment by PVince81** [comment]
:+1:

**Pull request merged**